src/Fixer/PhpTag/ShortEchoTagFixer.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\PhpTag;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use SplFileInfo;
+
+/**
+ * @author Michele Locati <michele@locati.it>
+ */
+final class ShortEchoTagFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /** @internal */
+    const OPTION_FORMAT = 'format';
+
+    /** @internal */
+    const OPTION_SHORT_ALWAYS = 'short_always';
+
+    /** @internal */
+    const OPTION_LONG_FUNCTION = 'long_function';
+
+    /** @internal */
+    const FORMAT_SHORT = 'short';
+
+    /** @internal */
+    const FORMAT_LONG = 'long';
+
+    /** @internal */
+    const LONG_FUNCTION_ECHO = 'echo';
+
+    /** @internal */
+    const LONG_FUNCTION_PRINT = 'print';
+
+    private $supportedFormatOptions = [
+        self::FORMAT_LONG,
+        self::FORMAT_SHORT,
+    ];
+
+    private $supportedLongFunctionOptions = [
+        self::LONG_FUNCTION_ECHO,
+        self::LONG_FUNCTION_PRINT,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        $sample = <<<'EOT'
+<?=1?>
+<?php print '2' . '3'; ?>
+<?php /* comment */ echo '2' . '3'; ?>
+<?php print '2' . '3'; someFunction(); ?>
+
+EOT
+        ;
+
+        return new FixerDefinition(
+            'Replace short-echo `<?=` with long format `<?php echo`/`<?php print` syntax, or vice-versa.',
+            [
+                new CodeSample($sample),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_LONG]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_LONG, self::OPTION_LONG_FUNCTION => self::LONG_FUNCTION_PRINT]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_SHORT]),
+                new CodeSample($sample, [self::OPTION_FORMAT => self::FORMAT_SHORT, self::OPTION_SHORT_ALWAYS => true]),
+            ],
+            null
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        if (self::FORMAT_SHORT === $this->configuration[self::OPTION_FORMAT]) {
+            if (null !== $tokens->findSequence([[\T_OPEN_TAG], [\T_ECHO]])) {
```

Comment:
In [Cookbook](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.16.0/doc/COOKBOOK-FIXERS.md) the method `isCandidate` has mentioning that it "has to be ultra cheap to execute" - I'm not sure if `findSequence` is fine to have here - @keradus, @SpacePossum, @julienfalque WDYT?

Reply:
If there's a faster way to check if `$tokens` contains `<?php echo` or `<?php print` I'd be glad to update the code...

Reply:
We don't have to search for the sequence - see the `isCandidate` method in other fixers - it could use `findSequence` in few of them, but the goal is the method is fast, not super accurate - I'm not saying this is right (maybe more complex `isCandidate` could save some files from being checked) - but it is the approach all other fixers follow.

Simple `$tokens->isAnyTokenKindsFound([T_ECHO, T_PRINT])` would be fine, I guess.

Reply:
You are absolutely right: 372f92e0f07fcb61ba78b83c4b5c918e4ae1ca69